### ISSUE TITLE
Debian fails to build with "deprecated declarations" error

### ISSUE THREAD

**Issue by agdphd** [limansky/tinymount#10]
The debian branch does not build under current Debian testing/unstable; the compiler returns deprecated declarations warnings, which are then treated as errors. Example log entries follow.

It probably isn't a permanent fix, but adding this line to debian/rules at least patches over the problem:

```
CPPFLAGS += -Wno-deprecated-declarations
```
## Error log

```
-> % dpkg-buildpackage
dpkg-buildpackage: source package tinymount
dpkg-buildpackage: source version 0.2.6
dpkg-buildpackage: source changed by Mike Limansky <mike.limansky@gmail.com>
dpkg-buildpackage: host architecture amd64
dpkg-source --before-build tinymount
fakeroot debian/rules clean
test -x debian/rules
dh_testroot
rm -f debian/stamp-makefile-build debian/stamp-makefile-install
/usr/bin/make  -C .  -k distclean
make[1]: Entering directory `/home/andrew/projects/build/tinymount'
cd src/ && /usr/bin/make -f Makefile distclean
make[2]: Entering directory `/home/andrew/projects/build/tinymount/src'
/usr/bin/make -f Makefile.Release distclean
make[3]: Entering directory `/home/andrew/projects/build/tinymount/src'
rm -f tinymount_ru.qm
rm -f release/moc_diskmanager.cpp release/moc_tinymounttray.cpp release/moc_settingsdialog.cpp release/moc_udisksinterface.cpp release/moc_udisksdeviceinterface.cpp
rm -f release/qrc_tinymount.cpp
rm -f ui_settingsdialog.h
rm -f release/main.o release/diskmanager.o release/tinymounttray.o release/settings.o release/settingsdialog.o release/udisksinterface.o release/udisksdeviceinterface.o release/libnotifier.o release/moc_diskmanager.o release/moc_tinymounttray.o release/moc_settingsdialog.o release/moc_udisksinterface.o release/moc_udisksdeviceinterface.o release/qrc_tinymount.o
rm -f *~ core *.core
rm -f tinymount 
rm -f Makefile.Release
make[3]: Leaving directory `/home/andrew/projects/build/tinymount/src'
/usr/bin/make -f Makefile.Debug distclean
make[3]: Entering directory `/home/andrew/projects/build/tinymount/src'
rm -f tinymount_ru.qm
rm -f debug/moc_diskmanager.cpp debug/moc_tinymounttray.cpp debug/moc_settingsdialog.cpp debug/moc_udisksinterface.cpp debug/moc_udisksdeviceinterface.cpp
rm -f debug/qrc_tinymount.cpp
rm -f ui_settingsdialog.h
rm -f debug/main.o debug/diskmanager.o debug/tinymounttray.o debug/settings.o debug/settingsdialog.o debug/udisksinterface.o debug/udisksdeviceinterface.o debug/libnotifier.o debug/moc_diskmanager.o debug/moc_tinymounttray.o debug/moc_settingsdialog.o debug/moc_udisksinterface.o debug/moc_udisksdeviceinterface.o debug/qrc_tinymount.o
rm -f *~ core *.core
rm -f tinymount 
rm -f Makefile.Debug
make[3]: Leaving directory `/home/andrew/projects/build/tinymount/src'
rm -f Makefile
make[2]: Leaving directory `/home/andrew/projects/build/tinymount/src'
rm -f Makefile
make[1]: Leaving directory `/home/andrew/projects/build/tinymount'
dh_clean 
rm -f ./Makefile ./.qmake.internal.cache
dpkg-source -b tinymount
dpkg-source: warning: no source format specified in debian/source/format, see dpkg-source(1)
dpkg-source: info: using source format `1.0'
dpkg-source: warning: source directory 'tinymount' is not <sourcepackage>-<upstreamversion> 'tinymount-0.2.6'
dpkg-source: info: building tinymount in tinymount_0.2.6.tar.gz
dpkg-source: info: building tinymount in tinymount_0.2.6.dsc
debian/rules build
test -x debian/rules
mkdir -p "."
cd . && qmake PREFIX=/usr CONFIG+=with_libnotify  'QMAKE_CC = cc' 'QMAKE_CXX = g++' 'QMAKE_CFLAGS_RELEASE = -D_FORTIFY_SOURCE=2 -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wall' 'QMAKE_CXXFLAGS_RELEASE = -D_FORTIFY_SOURCE=2 -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wall' 'QMAKE_LFLAGS_RELEASE = -Wl,-z,relro'
/usr/bin/make  -C .  
make[1]: Entering directory `/home/andrew/projects/build/tinymount'
cd src/ && /usr/bin/qmake /home/andrew/projects/build/tinymount/src/src.pro PREFIX=/usr CONFIG+=with_libnotify QMAKE_CC\ =\ cc QMAKE_CXX\ =\ g++ QMAKE_CFLAGS_RELEASE\ =\ -D_FORTIFY_SOURCE=2\ -g\ -O2\ -fstack-protector\ --param=ssp-buffer-size=4\ -Wformat\ -Werror=format-security\ -Wall QMAKE_CXXFLAGS_RELEASE\ =\ -D_FORTIFY_SOURCE=2\ -g\ -O2\ -fstack-protector\ --param=ssp-buffer-size=4\ -Wformat\ -Werror=format-security\ -Wall QMAKE_LFLAGS_RELEASE\ =\ -Wl,-z,relro -o Makefile
cd src/ && /usr/bin/make -f Makefile 
make[2]: Entering directory `/home/andrew/projects/build/tinymount/src'
/usr/bin/make -f Makefile.Release
make[3]: Entering directory `/home/andrew/projects/build/tinymount/src'
lrelease -silent translations/tinymount_ru.ts -qm tinymount_ru.qm
/usr/bin/uic-qt4 settingsdialog.ui -o ui_settingsdialog.h
g++ -c -m64 -pipe -Wall -Werror -D_FORTIFY_SOURCE=2 -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wall -pthread -Wall -W -D_REENTRANT -DQT_WEBKIT -DWITH_LIBNOTIFY -DTINYMOUNT_VERSION=\"0.2.6\" -DDATADIR=\"/usr/share/tinymount\" -DQT_NO_DEBUG -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4/QtDBus -I/usr/include/qt4 -I. -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libpng12 -Irelease -I. -o release/main.o main.cpp
main.cpp: In function ‘int main(int, char**)’:
main.cpp:135:5: error: ‘void g_type_init()’ is deprecated (declared at /usr/include/glib-2.0/gobject/gtype.h:669) [-Werror=deprecated-declarations]
main.cpp:135:17: error: ‘void g_type_init()’ is deprecated (declared at /usr/include/glib-2.0/gobject/gtype.h:669) [-Werror=deprecated-declarations]
cc1plus: all warnings being treated as errors
make[3]: *** [release/main.o] Error 1
make[3]: Leaving directory `/home/andrew/projects/build/tinymount/src'
make[2]: *** [release] Error 2
make[2]: Leaving directory `/home/andrew/projects/build/tinymount/src'
make[1]: *** [sub-src-make_default] Error 2
make[1]: Leaving directory `/home/andrew/projects/build/tinymount'
make: *** [debian/stamp-makefile-build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2
```

**Comment by limansky** [comment]
Thanks for report, I think better will be to remove -Werror on package building (as it currently done on Gentoo). I'll take care about this bug in very beginning of June.

**Comment by limansky** [comment]
Sorry for delay, I've fixed the warning in commit dfa794061f9ba060e5530baef9aa0db6a6e02b03, but still not changed Debianization.
